packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -155,8 +158,14 @@ export class Summarizer extends EventEmitter implements ISummarizer {
      * the run promise, and also close the container.
      * @param reason - reason code for stopping
      */
-    public stop(reason: SummarizerStopReason) {
+    public stop(reason: SummarizerStopReason, forceClose = false) {
         this.stopDeferred.resolve(reason);
+
+        // Should be used only by SummaryManager, when it knows we are not in run() method!
+        assert(this.running === forceClose, "forceClose");
```

Comment:
This means that forceClose can only be true when summarizer is already running? How can the caller be sure of that?

Reply:
Also, a more descriptive assert message might help :-).

Reply:
Weird, it should be the opposite. It's weird because I've run through UT and manual tests and did not hit this assert. I'd expect everything to fail. 
I maybe have screwed up with local builds, but official builds had only one failure (on rerun, I got two). Is our test coverage so bad?

Reply:
Hm, now I see failures (when I reversed it). Taking a look :)

Reply:
Hmmm. SummaryManager has its own state tracking and takes action based on that. Summarizer has its own state tracking now (this.running). This code is trying to assert that these two states are in sync. If the states become out of sync, we will hit this assert and maybe that's what's happening?

Reply:
I've added explicit close() method instead